src/components/ha-sidebar.ts

```diff
@@ -640,17 +701,91 @@ class HaSidebar extends LitElement {
     this._hideTooltip();
   }
 
-  private _listboxKeydown() {
+  private _getIndexOfTarget(evt: Event): number {
+    const listbox = evt.currentTarget as List;
+    const elements = listbox.items;
+    const path = evt.composedPath();
+
+    for (const pathItem of path as Node[]) {
+      let index = -1;
+      if (isNodeElement(pathItem) && isListItem(pathItem)) {
+        index = elements.indexOf(pathItem);
+      }
+
+      if (index !== -1) {
+        return index;
+      }
+    }
+
+    return -1;
```

Comment:
Why not use?
```suggestion
	return listbox.getIndexOfTarget(evt);	
```

Reply:
Because we use a extended `mwc-list-item` with a different name? Let's make an extended `mwc-list` that handles this. It doesn't belong in `ha-sidebar`.

Reply:
Yeah. That's probably for the best for a number of reasons, but going to be a non-trivial refactor. I'll get to that after Christmas then.